sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/DateTimeUtilsSuite.scala

```diff
@@ -623,10 +623,12 @@ class DateTimeUtilsSuite extends SparkFunSuite with Matchers with SQLHelper {
       "MIT" -> Set(15338))
     for (tz <- ALL_TIMEZONES) {
       val skipped = skipped_days.getOrElse(tz.getID, Set.empty)
-      (-20000 to 20000).foreach { d =>
+      val testingData = Seq(-20000, 20000) ++
+        (1 to 1000).map(_ => (math.random() * 40000 - 20000).toInt)
```

Comment:
There are 2 approaches for generating random values for Catalyst's types:
1. https://github.com/apache/spark/blob/1febd373ea806326d269a60048ee52543a76c918/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/LiteralGenerator.scala#L114-L124
2. RandomDataGenerator.forType(): https://github.com/apache/spark/blob/9562b26914f9a0c513e34b9b90cdec60067ef055/sql/catalyst/src/test/scala/org/apache/spark/sql/RandomDataGenerator.scala#L178-L193

Can you use one of them?

Reply:
Sorry seeing this comment too late.

Here we just need to get a random int. It's nothing about the catalyst data type. I think it's better to keep simple.

Reply:
It is not related to the PR but I think we could unify random data generators for Catalyst's types.